deepspeed/inference/engine.py

```diff
@@ -86,6 +86,27 @@ def build_bloom_alibi_tensor(attention_mask: torch.Tensor, num_heads: int, dtype
         return alibi.reshape(batch_size * num_heads, 1, seq_length).to(dtype)
 
 
+from typing import Optional
+
+
+def build_mpt_atten_bias_tensor(self,
+                                device,
+                                dtype,
+                                attention_mask: Optional[torch.ByteTensor] = None,
+                                prefix_mask: Optional[torch.ByteTensor] = None,
+                                sequence_id: Optional[torch.LongTensor] = None):
+    (attn_bias, attention_mask) = self._attn_bias_orig(device,
+                                                       dtype,
+                                                       attention_mask=attention_mask,
+                                                       prefix_mask=prefix_mask,
+                                                       sequence_id=sequence_id)
+    if dist.is_initialized():
+        num_heads_per_rank = int(self.config.n_heads / dist.get_world_size())
+        offset = dist.get_rank() * num_heads_per_rank
+        attn_bias = attn_bias[:, offset:num_heads_per_rank + offset, :, :]
+    return attn_bias, attention_mask
```

Comment:
The import should be at the top of the file with the rest of the imports. Also, could we please move this to wherever @RezaYazdaniAminabadi suggests? Thanks!

Reply:
@RezaYazdaniAminabadi should I place it to https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/module_inject/inject.py?

Reply:
Is it for model specific treatment of AutoTP?  I think a new file i.e. module_inject/auto_tp_model_utils.py to place all model specific policies might be a good place  @sywangyi  @RezaYazdaniAminabadi @mrwyattii